crates/bevy_scene/src/scene_spawner.rs

```diff
@@ -268,11 +268,25 @@ impl SceneSpawner {
         for (instance_id, parent) in scenes_with_parent {
             if let Some(instance) = self.spawned_instances.get(&instance_id) {
                 for entity in instance.entity_map.values() {
+                    // Add the `Parent` component to the scene root
                     if let Some(mut entity_mut) = world.get_entity_mut(entity) {
+                        // This will filter only the scene root entity, as all other from the
+                        // scene have a parent
                         if !entity_mut.contains::<Parent>() {
                             entity_mut.insert(Parent(parent));
                         }
                     }
+                    // Add the scene root to the `Children` component on the parent
+                    if let Some(mut parent_entity) = world.get_entity_mut(parent) {
+                        if let Some(children) = parent_entity.get_mut::<Children>() {
+                            let children = &**children;
+                            let mut children = children.to_vec();
+                            children.push(entity);
+                            parent_entity.insert(Children::with(&children));
```

Comment:
This seems like quite a bit of vector coppying.
Is it possible to just push `entity` to the parent's children component?

Reply:
the `SmallVec` inside `Children` is not mutably accessible because it caused a lot of sync issues:
https://github.com/bevyengine/bevy/blob/46b822e3da25bac8affe17e946e61fc1c07500f1/crates/bevy_transform/src/components/children.rs#L11

Reply:
I think it would be nice to add a comment about this?